mailpoet/lib/Newsletter/Scheduler/AutomationEmailScheduler.php

```diff
@@ -75,7 +75,19 @@ public function getScheduledTaskSubscriber(NewsletterEntity $email, SubscriberEn
       ->setParameter('newsletter', $email)
       ->setParameter('subscriber', $subscriber)
       ->getQuery()
-      ->getOneOrNullResult();
+      ->getResult();
+    $result = null;
+    foreach ($results as $scheduledTaskSubscriber) {
+      $task = $scheduledTaskSubscriber->getTask();
+      if (!$task instanceof ScheduledTaskEntity) {
+        continue;
+      }
+      $meta = $task->getMeta();
+      if (($meta['automation']['run_id'] ?? null) === $runId) {
+        $result = $scheduledTaskSubscriber;
+        break;
+      }
+    }
```

Comment:
@costasovo Maybe I'm overthinking this, but what if there are many of such records? Won't that be suboptimal in terms of always fetching all of them? I don't have a specific scenario, but an automation can be anything... I could imagine that you send an email after every order or something, and then, if a subscriber reaches hundreds of orders, we could run into performance issues or memory errors here.

I know that addressing this properly is not easy given the `meta` field usage (would need a more complex query matching the run ID on the DBAL level).

But I have another idea — what if we limit the scheduled tasks so that `$automationRun->getCreatedAt()` is `less or equal` than scheduled task's (or sending queue's) `createdAt`?

Reply:
@JanJakes Thanks for noticing the potential performance issue. I'll update the PR based on your suggestions.